app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -45,6 +49,197 @@ def __init__(self, schema):
+    id: int
+    name: str
+    slug: str
+    description: str
+    application: str
+    ownerEmail: str
+    schema: str
+
+
+@dataclass
+class UserDataClass:
+    username: str
+
+
+@dataclass
+class TargetingConfigDataClass:
+    label: str
+    value: str
+    applicationValues: typing.List[str]
+    description: str
+
+
+@dataclass
+class MetricDataClass:
+    slug: str
+    friendlyName: str
+    description: str
+
+
+@dataclass
+class OutcomeDataClass:
+    application: str
+    description: str
+    friendlyName: str
+    slug: str
+    isDefault: bool
+    metrics: typing.List[MetricDataClass]
+
+
+@dataclass
+class NimbusConfigurationDataClass:
+    applications: typing.List[LabelValueDataClass]
+    channels: typing.List[LabelValueDataClass]
+    applicationConfigs: typing.List[ApplicationConfigDataClass]
+    countries: typing.List[GeoDataClass]
+    locales: typing.List[GeoDataClass]
+    languages: typing.List[GeoDataClass]
+    documentationLink: typing.List[LabelValueDataClass]
+    allFeatureConfigs: typing.List[FeatureConfigDataClass]
+    firefoxVersions: typing.List[LabelValueDataClass]
+    outcomes: typing.List[OutcomeDataClass]
+    owners: typing.List[UserDataClass]
+    targetingConfigs: typing.List[TargetingConfigDataClass]
+    conclusionRecommendations: typing.List[LabelValueDataClass]
+    hypothesisDefault: str = NimbusExperiment.HYPOTHESIS_DEFAULT
+    maxPrimaryOutcomes: int = NimbusExperiment.MAX_PRIMARY_OUTCOMES
+
+    def __init__(self):
+        self.applications = self._enum_to_label_value(NimbusExperiment.Application)
+        self.channels = self._enum_to_label_value(NimbusExperiment.Channel)
+        self.applicationConfigs = self._get_application_configs()
+        self.countries = self._geo_model_to_dataclass(
+            Country.objects.all().order_by("name")
+        )
+        self.locales = self._geo_model_to_dataclass(Locale.objects.all().order_by("name"))
+        self.languages = self._geo_model_to_dataclass(
+            Language.objects.all().order_by("name")
+        )
+        self.documentationLink = self._enum_to_label_value(
+            NimbusExperiment.DocumentationLink
+        )
+        self.allFeatureConfigs = self._get_feature_configs()
+        self.firefoxVersions = self._enum_to_label_value(NimbusExperiment.Version)
+        self.outcomes = self._get_outcomes()
+        self.owners = self._get_owners()
+        self.targetingConfigs = self._get_targeting_configs()
+        self.conclusionRecommendations = self._enum_to_label_value(
+            NimbusExperiment.ConclusionRecommendation
+        )
+
+    def _geo_model_to_dataclass(self, queryset):
+        return [GeoDataClass(id=i.id, name=i.name, code=i.code) for i in queryset]
+
+    def _enum_to_label_value(self, text_choices):
+        return [
+            LabelValueDataClass(
+                label=text_choices[name].label,
+                value=name,
+            )
+            for name in text_choices.names
+        ]
```

Comment:
Functions inside of a dataclass? I'm curious to know you're thoughts behind this.

Reply:
I don't follow what you mean?